src/LiveComponent/src/Twig/DeterministicTwigIdCalculator.php

```diff
@@ -56,8 +52,125 @@ public function calculateDeterministicId(): string
             crc32($fileAndLine),
             $this->lineAndFileCounts[$fileAndLine]
         );
-        ++$this->lineAndFileCounts[$fileAndLine];
+
+        if ($increment) {
+            ++$this->lineAndFileCounts[$fileAndLine];
+        }
 
         return $id;
     }
+
+    public function reset(): void
+    {
+        $this->lineAndFileCounts = [];
+    }
+
+    /**
+     * Adapted from Twig\Error\Error::guessTemplateInfo().
+     *
+     * Any differences are marked below.
+     *
+     * @return array{name: string, line: int}
+     */
+    private function guessTemplateInfo(): array
```

Comment:
I didn't think it possibly to make this _more_ complex! 😆

Reply:
whoooooops :p 

I'm pretty excited to have Twig in Symfony... maybe this is yet another thing that could be "fixed"/ made available.